pkg/controller/daemonset/daemonset_util.go

```diff
@@ -282,7 +282,7 @@ func storeDaemonSetStatus(dsClient kubeClient.Client, ds *appsv1alpha1.DaemonSet
 	klog.V(6).Infof("toUpdate is %v", ds)
 
 	var updateErr, getErr error
-	for i := 0; i < StatusUpdateRetries; i++ {
+	for i := 0; ; i++ {
```

Comment:
I suggest to remove the StatusUpdateRetries and use the client-go retry package. i.e., the retry.RetryOnConflict method.